crate/agent/src/main.rs

```diff
@@ -108,3 +86,51 @@ fn init_logging() {
         .with(stdout_layer)
         .init();
 }
+
+fn initialize_agent(conf: &CosmianVmAgent) -> Result<()> {
+    let ssl_private_key = &conf.agent.ssl_private_key;
+    let ssl_certificate = &conf.agent.ssl_certificate;
+
+    // Generate the certificate if not present
+    match (ssl_private_key.exists(), ssl_certificate.exists()) {
+        (false, false) => {
+            tracing::info!("Generating certificates...");
+            let hostname = gethostname();
+            let hostname = hostname.to_string_lossy();
+            let subject = format!("CN={hostname},O=Cosmian Tech,C=FR,L=Paris,ST=Ile-de-France");
+            let (sk, cert) = generate_self_signed_cert(
+                &subject,
+                &[&conf.agent.host],
+                TLS_DAYS_BEFORE_EXPIRATION,
+            )?;
+
+            let mut file = File::create(ssl_certificate)?;
+            file.write_all(cert.as_bytes())?;
+
+            let mut file = File::create(ssl_private_key)?;
+            file.write_all(sk.as_bytes())?;
```

Comment:
```suggestion
            std::fs::write(ssl_certificate, cert)?;
            std::fs::write(ssl_private_key, sk)?;
```

also, don't forget to remove imports